File: typescript/ajustado_com_pooling/src/app/common/errors/base.error.ts
import { HttpResponseError } from "../interfaces/http.interfaces";

export class BaseError {
  protected _message: string;
  protected _statusCode: number = 500;
  protected _errors?: string;

  constructor(message: string, statusCode: number, errors?: string) {
    this._message = message;
    this._statusCode = statusCode;
    this._errors = errors;
  }

  get message() {
    return this._message;
  }

  get statusCode() {
    return this._statusCode;
  }

  get errors() {
    return this._errors;
  }

  toJSON(): HttpResponseError {
    return {
      statusCode: this._statusCode,
      message: this._message,
      errors: this?._errors,
    };
  }
}

File: python/flask/src/app/common/errors/base_error.py
from werkzeug.exceptions import HTTPException
from pydantic import ValidationError
from ..messages import MESSAGES

class BaseError(HTTPException):
  code = 500
  description = MESSAGES["error"]["InternalServer"]

  def __init__(self, code: int = None, message: str = None, errors: str = None):
        if code is not None:
            self.code = code
        if message is None:
            message = self.description

        self.response = {
            "statusCode": self.code,
            "message": message,
        }

        if isinstance(errors, ValidationError):
            errors = [f"{err['loc'][0]}: {err['msg']}" for err in errors.errors()]

        if errors is not None:
            self.response['errors'] = errors

        super().__init__(description=self.response)
